.tools/validation/metadata.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python3
+
+from dataclasses import dataclass, field
+from typing import Optional, Self
+from os.path import splitext
+import metadata_errors
+from metadata_errors import MetadataErrors, MetadataParseError, DuplicateItemException
+from doc_gen import DocGen
+
+
+@dataclass
+class Url:
+    title: str
+    url: Optional[str]
+
+    @staticmethod
```

Comment:
Because the method doesn't need access to the `cls` object?

Reply:
I suppose I think of classmethod as semantically implying it is a factory method, like what you're doing here. So you'd return cls(title, url) instead of Url(title,url), which to me feels cleaner. Up to you.

Reply:
Thanks I hate it - `grep 'Version('` should give me all constructors, which that does not.